lib/internal/process.js

```diff
@@ -22,6 +23,53 @@ const assert = process.assert = function(x, msg) {
 };
 
 
+// Set up the process.cpuUsage() function.
+function setup_cpuUsage() {
+  // Get the native function, which will be replaced with a JS version.
+  const _cpuUsage = process.cpuUsage;
+
+  // Create the argument array that will be passed to the native function.
+  const cpuValues = new Float64Array(2);
+
+  // Replace the native function with the JS version that calls the native
+  // function.
+  process.cpuUsage = function cpuUsage(prevValue) {
+    // If a previous value was passed in, ensure it has the correct shape.
+    if (prevValue) {
+      if (!previousValueIsValid(prevValue.user)) {
+        throw new TypeError('value of user property of argument is invalid');
+      }
+
+      if (!previousValueIsValid(prevValue.system)) {
+        throw new TypeError('value of system property of argument is invalid');
+      }
+    }
+
+    // Call the native function to get the current values.
+    const errmsg = _cpuUsage(cpuValues);
+    if (errmsg) {
+      throw new Error('unable to obtain CPU usage: ' + errmsg);
+    }
+
+    // If a previous value was passed in, return diff of current from previous.
+    if (prevValue) return {
+      user: cpuValues[0] - prevValue.user,
+      system: cpuValues[1] - prevValue.system
+    };
+
+    // If no previous value passed in, return current value.
+    return {
+      user: cpuValues[0],
+      system: cpuValues[1]
+    };
+
+    function previousValueIsValid(num) {
+      return Number.isFinite(num) && num >= 0;
```

Comment:
Since we're fine with floating point values, in case we opt for nanosecond level precision in the future, I'd prefer the check to be:
```js
return num <= Number.MAX_SAFE_INTEGER && num >= 0;
```
That will grab `NaN`, and verify the value is within proper bounds (since anything over 2^53 can't be a valid value). It will allow string -> number coercion for things like `'0xff'`, but I don't think that's a concern.